rllib/models/specs/tests/test_spec_dict.py

```diff
@@ -0,0 +1,189 @@
+import unittest
+import numpy as np
+
+from ray.rllib.models.specs.specs_np import NPTensorSpec
+from ray.rllib.models.specs.specs_dict import SpecDict
+from ray.rllib.models.specs.checker import check_input_specs
+
+
+class TypeClass1:
+    pass
+
+
+class TypeClass2:
+    pass
+
+
+class TestSpecDict(unittest.TestCase):
+    def test_basic_validation(self):
+
+        h1, h2 = 3, 4
+        spec_1 = SpecDict(
+            {
+                "out_tensor_1": NPTensorSpec("b, h", h=h1),
+                "out_tensor_2": NPTensorSpec("b, h", h=h2),
+                "out_class_1": TypeClass1,
+            }
+        )
+
+        # test validation.
+        tensor_1 = {
+            "out_tensor_1": np.random.randn(2, h1),
+            "out_tensor_2": np.random.randn(2, h2),
+            "out_class_1": TypeClass1(),
+        }
+
+        spec_1.validate(tensor_1)
+
+        # test missing key in specs
+        tensor_2 = {
+            "out_tensor_1": np.random.randn(2, h1),
+            "out_tensor_2": np.random.randn(2, h2),
+        }
+
+        self.assertRaises(ValueError, lambda: spec_1.validate(tensor_2))
+
+        # test missing key in data
```

Comment:
nit: "Test additional key in tensor (not mentioned in spec)."

Reply:
fixed.